docs/src/modules/components/Demo.js

```diff
@@ -473,7 +473,9 @@ export default function Demo(props) {
         `The following demos use TS directly: ${demoOptions.demo}.`,
         '',
         'Please run "pnpm docs:typescript:formatted" to generate a JS version and reference it:',
-        `{{"demo": "${demoOptions.demo.replace(/\.(.*)$/, '.js')}", …}}.`,
```

Comment:
I got curious, I looked closer at this. I see no catastrophic backtracking here https://www.regular-expressions.info/catastrophic.html but it's interesting how it's faster when there are a lot of dots:

before
<img width="560" alt="SCR-20241216-tgxa" src="https://github.com/user-attachments/assets/9116e78c-7f35-4e1e-94a1-8c82f3152260" />

after:
<img width="562" alt="SCR-20241216-tgyb" src="https://github.com/user-attachments/assets/d8b69129-52bb-4891-976c-6f1962c9e8f5" />

Now, in practice, this doesn't matter.

Reply:
Regex101 is suitable for learning regexes and testing the functionality of regexes in development. However, in serious software engineering and programming theory research, its behavior often deviates from the true regex engines behind the languages it lists. Additionally, from the image you provided, since it shows step numbers, I can infer that you might be using PHP's PCRE engine rather than JavaScript's regex engine in Regex101, which also has a certain degree of imprecision (perhaps I can try to provide a convenient external access to my instrumentation for migration counts of bytecode in real V8's regex engine, but it may take some time). For this regex, believe what you see; you can use a very simple example to confirm whether there is an efficiency issue:

```javascript
const str = "." + "a".repeat(100000) + "\n@.ts";
console.time('plain string');
const test1 = str.replace(/\.(.*)$/, '.js');
console.timeEnd('plain string');

const attackStr = "." + ".".repeat(100000) + "\n@.ts";
console.time('attack string');
const test2 = attackStr.replace(/\.(.*)$/, '.js');
console.timeEnd('attack string');
```

On my PC, it shows the following results:

```
plain string: 0.174072265625 ms
attack string: 2774.43994140625 ms
```

It easily demonstrates a performance gap of about 15,000 times. Preemptively making some adjustments to avoid this does not seem harmful as long as it does not affect actual functionality.

Regarding your statement that no catastrophic backtracking was found in this case, this example differs slightly from those on the web page you referenced; its backtracking behavior comes from JavaScript's default "substring matching" feature. That is to say, all matching behaviors of regex `<r>` in JavaScript can be approximately equivalent to matching `^.*<r>$` in engines like Java or Python. In this example, `.*\.(.*)` forms a pathological sub regex where the engine cannot distinguish whether each dot belongs to the first or second `.*`, leading to numerous attempts and backtracking. [Some studies](https://www.usenix.org/system/files/sec21-li-yeting.pdf) have categorized this as "Starting with Large Quantifier (SLQ)" pattern.

However, I see that this file is named `Demo.js`, so I understand why you think this issue would not cause much impact in your project. But if making a simple modification does not affect your functionality and can prevent others from potentially misusing or referencing code snippets unintentionally causing issues down the line, I still believe it benefits the software ecosystem overall without any harm done. Ultimately though, you are members of this project and I respect all decisions regarding how you handle your project.